src/Domains/Social/Messages/Workflows/Activities/DefaultMessageActivity.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kanvas\Social\Messages\Workflows\Activities;
+
+use Baka\Contracts\AppInterface;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Kanvas\Social\Messages\Actions\CreateMessageAction;
+use Kanvas\Social\Messages\DataTransferObject\MessageInput;
+use Kanvas\Social\MessagesTypes\Actions\CreateMessageTypeAction;
+use Kanvas\Social\MessagesTypes\DataTransferObject\MessageTypeInput;
+use Kanvas\Social\MessagesTypes\Repositories\MessagesTypesRepository;
+use Kanvas\Workflow\Contracts\WorkflowActivityInterface;
+use Workflow\Activity;
+
+class DefaultMessageActivity extends Activity implements WorkflowActivityInterface
+{
+    public function execute(Model $entity, AppInterface $app, array $params): array
+    {
+        try {
+            $messageType = MessagesTypesRepository::getByVerb('system-message', $app);
+        } catch (ModelNotFoundException $e) {
+            $messageTypeDto = MessageTypeInput::from([
+                'apps_id' => $app->getId(),
+                'name' => 'system-message',
+                'verb' => 'system-message',
+            ]);
+            $messageType = (new CreateMessageTypeAction($messageTypeDto))->execute();
+        }
+        $messages = [];
+        foreach ($entity->getAllCustomFields() as $customField) {
```

Comment:
@FredPeal instead of using all the custom field to create msg, specify a field or array of field to create msg on the workflow configuration similar to this https://github.com/bakaphp/kanvas-ecosystem-api/blob/development/src/Domains/Connectors/Internal/Activities/UserCustomFieldActivity.php#L22